lib/r10k/action/runner.rb

```diff
@@ -44,10 +44,13 @@ def setup_settings
 
         overrides = {}
         overrides[:cachedir] = @opts[:cachedir] if @opts.key?(:cachedir)
-        overrides[:deploy] = {} if @opts.key?(:'puppet-path') || @opts.key?(:'generate-types')
-        overrides[:deploy][:puppet_path] = @opts[:'puppet-path'] if @opts.key?(:'puppet-path')
-        overrides[:deploy][:puppet_conf] = @opts[:'puppet-conf'] unless @opts[:'puppet-conf'].nil?
-        overrides[:deploy][:generate_types] = @opts[:'generate-types'] if @opts.key?(:'generate-types')
+        if @opts.key?(:'puppet-path') || @opts.key?(:'generate-types') || @opts.key?(:'deploy-spec')
+          overrides[:deploy] = {}
+          overrides[:deploy][:puppet_path] = @opts[:'puppet-path'] if @opts.key?(:'puppet-path')
+          overrides[:deploy][:puppet_conf] = @opts[:'puppet-conf'] unless @opts[:'puppet-conf'].nil?
```

Comment:
Is there a reason you're using `unless` and `.nil?` for this instead of what you do for all the other options?
```suggestion
          overrides[:deploy][:puppet_conf] = @opts[:'puppet-conf'] if @opts[:'puppet-conf']
```

Reply:
I guess that's how it was before..

Reply:
This would make sense if we ever expected `puppet-conf` to be deliberately set to `false`. But that seems kind of unlikely. Looks like this one was added recently by a [community member](https://github.com/puppetlabs/r10k/commit/3a922859ba0ee01db25999ebc6e94fa0f3078d80) who didn't bother to follow the pattern of the others, so I imagine it could be changed to use the `if`.

Reply:
Looks like the point of that PR was to allow specifying a config for use with `generate type` (which is enabled with another of these options) so it's not surprising this wasn't added to the check: it was not being used on its own.